backend/open_webui/utils/middleware.py

```diff
@@ -1356,6 +1356,27 @@ async def stream_body_handler(response):
                         try:
                             data = json.loads(data)
 
+                            extra_params = {
+                                "__event_emitter__": get_event_emitter(metadata),
+                                "__event_call__": get_event_call(metadata),
+                                "__user__": {
+                                    "id": user.id,
+                                    "email": user.email,
+                                    "name": user.name,
+                                    "role": user.role,
+                                },
+                                "__metadata__": metadata,
+                                "__request__": request,
```

Comment:
User might also wish to have more fine grain control over content rendering. We can further add `__content__` and `__ content_blocks__` to `extra_params`. Then user Filter can read from data, parse data and modify `content` and `content_block`, then simply remove delta from data.